update-extension-dependencies.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Purpose: Updates pom.xml files with symbolic dependencies to extensions to enforce a consistent build order.
+
+set -e -u -o pipefail
+shopt -s failglob
+
+echo ''
+echo 'Building bom-descriptor-json...'
+echo ''
+#mvn -D quickly -T0.8C -f independent-projects/arc
+#mvn -D quickly -T0.8C -f bom/application
+#mvn -D quickly -T0.8C -f core
+#mvn -D quickly -T0.8C -f extensions
+mvn clean package -f devtools/bom-descriptor-json
+
+DEP_TEMPLATE='        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>XXX</artifactId>
+            <version>\${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>'
+
+echo ''
+echo 'Building dependencies list from bom-descriptor-json...'
+echo ''
+DEPS=`grep -Po '(?<="artifact-id": ")[^"]+' devtools/bom-descriptor-json/target/*.json \
+  | sort \
+  | xargs -i sh -c "echo \"${DEP_TEMPLATE}\" | sed 's/XXX/{}/'" \
+  | sed ':a;N;$!ba;s/\n/\\\n/g'`    # replace newlines with \n so that sed accepts ${DEPS} as input
+
+MARK_START='<!-- START update-extension-dependencies.sh -->'
+MARK_END='<!-- END update-extension-dependencies.sh -->'
+
+echo ''
+echo 'Updating bom-descriptor-json/pom.xml...'
+echo ''
+sed -i "/${MARK_START}/,/${MARK_END}/c\        ${MARK_START}\n${DEPS}\n        ${MARK_END}" devtools/bom-descriptor-json/pom.xml
```

Comment:
More pom.xml could be added here although I'm asking myself is whether `docs` should really receive the very same deps as `bom-descriptor-json`?

Reply:
I've just pushed an update that includes `docs`. 🤷‍♂️